Content.Server/Salvage/SalvageSystem.cs

```diff
@@ -0,0 +1,274 @@
+                    break;
+                case SalvageSystemState.LettingGo:
+                    args.PushMarkup(Loc.GetString("salvage-system-magnet-examined-letting-go"));
+                    break;
+            }
+        }
+
+        private void Reset(RoundRestartCleanupEvent ev)
+        {
+            PulledObject = EntityUid.Invalid;
+            State = SalvageSystemState.Inactive;
+            StateTimer = 0.0f;
+        }
+
+        private void OnInteractHand(EntityUid uid, SalvageMagnetComponent sm, InteractHandEvent args)
+        {
+            if (args.Handled)
+                return;
+            args.Handled = true;
+            _popupSystem.PopupEntity(CallSalvage(), uid, Filter.Entities(args.User.Uid));
+        }
+
+        private MapCoordinates GetSalvagePlacementLocation()
+        {
+            foreach (var (smc, tsc) in EntityManager.EntityQuery<SalvageMagnetComponent, TransformComponent>(true))
+            {
+                return (new EntityCoordinates(smc.OwnerUid, smc.Offset).ToMap(EntityManager));
+            }
+            return MapCoordinates.Nullspace;
+        }
+
+        private IEnumerable<SalvageMapPrototype> GetAllSalvageMaps()
+        {
+            return _prototypeManager.EnumeratePrototypes<SalvageMapPrototype>();
+        }
+
+        // String is announced
+        private string SpawnSalvage()
+        {
+            // In case of failure
+            State = SalvageSystemState.Inactive;
+
+            var spl = GetSalvagePlacementLocation();
+            if (spl == MapCoordinates.Nullspace)
+            {
+                return Loc.GetString("salvage-system-announcement-spawn-magnet-lost");
+            }
+
+            SalvageMapPrototype? map = null;
+
+            var forcedSalvage = _configurationManager.GetCVar<string>(CCVars.SalvageForced);
+            if (forcedSalvage == "")
+            {
+                var allSalvageMaps = GetAllSalvageMaps().ToList();
+
+                for (var i = 0; i < allSalvageMaps.Count; i++)
+                {
+                    map = _random.PickAndTake(allSalvageMaps);
+
+                    if (_physicsSystem.TryCollideRect(Box2.CenteredAround(spl.Position, new Vector2(map.Size * 2.0f, map.Size * 2.0f)), spl.MapId, false))
+                    {
+                        // collided: set map to null so we don't spawn it
+                        map = null;
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+            }
+            else
+            {
+                _prototypeManager.TryIndex<SalvageMapPrototype>(forcedSalvage, out map);
+            }
+
+            if (map == null)
+            {
+                return Loc.GetString("salvage-system-announcement-spawn-no-debris-available");
+            }
+
+            var bp = _mapLoader.LoadBlueprint(spl.MapId, map.MapPath);
```

Comment:
Due to a limit in map loading rn you should load this in nullspace and then teleport it into place.

Reply:
wdym by "limit in maploading"? if you are referring to streaming, placing the map into nullspace first wont remidy this at its gonna be streamed in regardless only when it enters pvs

Reply:
I tried loading it in nullspace. Crashed.

Reply:
Need to make sure the station never spawns anywhere near 0,0 then